lib/hexpm_web/controllers/package_report_controller.ex

```diff
@@ -0,0 +1,146 @@
+defmodule HexpmWeb.PackageReportController do
+    use HexpmWeb, :controller
+
+    @package_reports_per_page 30
+    @sort_params ~w(timestamp)
+    @new_report_msg "Package report generated"
+    @report_updated_msg "Package report updated"
+
+    def index(conn, params) do
+        page_param = Hexpm.Utils.safe_int(params["page"]) || 1
+        reports = fetch_package_reports(@package_reports_per_page, page_param, conn.assigns.current_user)
+        reports_count = Enum.count(reports)
+        page = Hexpm.Utils.safe_page(page_param, reports_count, @packages_per_page)
+        
+        
+        render(
+            conn,
+            "index.html",
+            reports: reports,
+            per_page: @package_reports_per_page,
+            page: page,
+            total: reports_count
+        )
+    end
+
+    def new(conn,params) do
+        package = params["package"]
+
+        if conn.assigns.current_user do
+            if package do
+                build_report_form(conn, params)
+            else
+                fail_with(conn, "No package selected.")
+            end
+        else
+            fail_with(conn, "Need to be logged to submit a report.")
+        end
+    end
+    
+    def create(conn, params) do
+        description = params["description"]
+        package_name = params["package"]
+        state = "to_accept"
+        from_version = params["from_version"]
+        to_version = params["to_version"]
```

Comment:
A version requirement would be preferable to a from_version / to_version. Then you can use the functions in https://hexdocs.pm/elixir/Version.html.